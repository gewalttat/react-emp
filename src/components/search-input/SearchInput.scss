.search-input-panel {
  position: absolute;
  left: 178px;
  top: 150px;
}

.search-input {
  width: 800px;
  height: 57px;
  background: rgba(50, 50, 50, 0.8);
  mix-blend-mode: normal;
  opacity: 0.7;
  border: none;
  border-radius: 4px;
  font-family: Montserrat, sans-serif;
  font-style: normal;
  font-weight: 300;
  font-size: 20px;
  line-height: 24px;
  padding-left: 20px;
  margin-left: -30px;
  color: #ffffff;
  outline: none;
}

.search-button {
  margin-top: -57px;
  margin-left: 810px;
  width: 233px;
  height: 57px;
  background: #f65260;
  border-radius: 4px;
  display: flex;
  align-items: center;
  justify-content: center;

  &:hover {
    cursor: pointer;
    background: #f33242;
  }
}

.search-button-caption {
  font-family: Montserrat, sans-serif;
  font-style: normal;
  font-weight: 300;
  font-size: 20px;
  line-height: 24px;
  text-align: right;
  text-transform: uppercase;
  color: #ffffff;
}

.search-input-header {
  padding-bottom: 40px;
  margin-left: -30px;
  font-family: Montserrat, sans-serif;
  font-style: normal;
  font-weight: 300;
  font-size: 40px;
  line-height: 49px;
  text-transform: uppercase;
  color: #ffffff;
}