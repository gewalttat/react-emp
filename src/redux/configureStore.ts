/* eslint-disable @typescript-eslint/ban-ts-comment */
import rootReducer, { rootSaga } from './rootReducer';
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import createSagaMiddleware, { END } from 'redux-saga';

const sagaMiddleware = createSagaMiddleware();

const createStore = () => {
  const store = configureStore({
    reducer: rootReducer,
    middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(sagaMiddleware),
  });

  // @ts-ignore
  store.runSaga = () => sagaMiddleware.run(rootSaga);
  // @ts-ignore
  store.close = () => store.dispatch(END);

  return store;
};

export default createStore;