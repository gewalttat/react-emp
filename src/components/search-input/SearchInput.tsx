import React, { FC, useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import { searchMovie } from '../../redux/moviesReducer';
import { useHistory } from 'react-router';
import './SearchInput.scss';

export const SearchInput: FC = () => {
    const [searchInput, setSearchInput] = useState<string>('');
    const dispatch = useDispatch();
    const history = useHistory();

    const handleSearchInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setSearchInput(event.target.value);
    };

    useEffect(() => {
        dispatch(searchMovie(searchInput));
        history.push({
            pathname: '/search',
            search: `?${searchInput}`
        });
    }, [searchInput])

    return (
        <>
            <div className='search-input-panel'>
                <div className='search-input-header'>
                    <span>find your movie</span>
                </div>
                <div>
                    <input
                        type="text"
                        placeholder='What do you want to watch?'
                        value={searchInput}
                        onChange={handleSearchInputChange}
                        className='search-input' />
                    <div
                        className='search-button'
                        onClick={() => console.log('search event')}>
                        <span className='search-button-caption'>search</span>
                    </div>
                </div>
            </div>
        </>
    )
}