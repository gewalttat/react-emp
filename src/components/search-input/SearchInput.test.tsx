import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import {SearchInput} from './SearchInput';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

describe('With React Testing Library', () => {
  const initialState = {output:10}
  const mockStore = configureStore();

  it('input renders correctly', () => {
    const store = mockStore(initialState);
    const tree = renderer.create(<Provider store={store}><SearchInput/></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows find your movie', () => {
    const store = mockStore(initialState);
    const {getByText} = render(<Provider store={store}><SearchInput/></Provider>);
    expect(getByText('find your movie')).not.toBeNull();
  });

  it('shows search button', () => {
    const store = mockStore(initialState);
    const {getByText} = render(<Provider store={store}><SearchInput/></Provider>);
    expect(getByText('search')).not.toBeNull();
  });

  it('shows input changes correctly', () => {
    const store = mockStore(initialState);
    const setupInput = () => {
      const utils = render(<Provider store={store}><SearchInput/></Provider>)
      const input = utils.getByPlaceholderText('What do you want to watch?');
      return {
        input,
        ...utils,
      }
    }
    const {input} = setupInput();
    fireEvent.change(input, {target: {value: 'Fifty'}});
    expect(input.value).toBe('Fifty');
  })
})