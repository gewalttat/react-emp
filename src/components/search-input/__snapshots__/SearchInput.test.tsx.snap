// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With React Testing Library input renders correctly 1`] = `
<div
  className="search-input-panel"
>
  <div
    className="search-input-header"
  >
    <span>
      find your movie
    </span>
  </div>
  <div>
    <input
      aria-label="cost-input"
      className="search-input"
      onChange={[Function]}
      placeholder="What do you want to watch?"
      type="text"
      value=""
    />
    <div
      className="search-button"
    >
      <span
        className="search-button-caption"
        onClick={[Function]}
      >
        search
      </span>
    </div>
  </div>
</div>
`;

exports[`With React Testing Library renders correctly 1`] = `
<div
  className="search-input-panel"
>
  <div
    className="search-input-header"
  >
    <span>
      find your movie
    </span>
  </div>
  <div>
    <input
      aria-label="cost-input"
      className="search-input"
      onChange={[Function]}
      placeholder="What do you want to watch?"
      type="text"
      value=""
    />
    <div
      className="search-button"
    >
      <span
        className="search-button-caption"
        onClick={[Function]}
      >
        search
      </span>
    </div>
  </div>
</div>
`;

exports[`With React Testing Library search button renders correctly 1`] = `
<div
  className="search-input-panel"
>
  <div
    className="search-input-header"
  >
    <span>
      find your movie
    </span>
  </div>
  <div>
    <input
      aria-label="cost-input"
      className="search-input"
      onChange={[Function]}
      placeholder="What do you want to watch?"
      type="text"
      value=""
    />
    <div
      className="search-button"
    >
      <span
        className="search-button-caption"
        onClick={[Function]}
      >
        search
      </span>
    </div>
  </div>
</div>
`;
